src/crypto/index.js

```diff
@@ -856,12 +856,13 @@ Crypto.prototype.requestRoomKey = function(requestBody, recipients) {
  *
  * @param {module:crypto~RoomKeyRequestBody} requestBody
  *    parameters to match for cancellation
+ * @returns {Promise}
  */
 Crypto.prototype.cancelRoomKeyRequest = function(requestBody) {
-    this._outgoingRoomKeyRequestManager.cancelRoomKeyRequest(requestBody)
+    return this._outgoingRoomKeyRequestManager.cancelRoomKeyRequest(requestBody)
     .catch((e) => {
         console.warn("Error clearing pending room key requests", e);
```

Comment:
Are you sure we want to swallow the error for your usecase?

Reply:
If `cancelRoomKeyRequest`, the request was in an unhandled state or it didn't find a matching request in the store. If there's no matching request then there's nothing to cancel, so it's fine?